### DISCUSSION TITLE
Heltec iot board, 64bit ARM, Lora/wifi/BT, any shortcomings?

### DISCUSSION THREAD

**Discussion by pizzadoglover911** [markqvist/Reticulum#281] [General]
Hey folks,

I stumbled about this: https://heltec.org/project/sufficient-iot-hub/ board on heltec website. 64 bit arm processor, Linux distro tailored for it, Lora and wifi/Ethernet. 

 It seems like an "Rnode" slam dunk to me!

 But some things are too good to be true. Any reason this would fall short trying to operate RNS and lora functions? 

Some thoughts I have:
-(reticulum service) Which serial port would the sx1262 be referenced on? (For use as a reticulum rnode interface)

-(rnode): would this be the same as supporting any other sx1262 Lora board?

-(rns)How could one tell if the Linux build supports dependencies? 

-(hardware)Could this function as a replacement for both a raspberry pi and lora board combination?       
-Lower power usage? 
-Consolidated i/o ports. 

What do you guys think? I may buy one to try if the idea has any legs to stand on...

**Comment by Swissbandit** [comment]
that is one of the promising new boards, but we are not quite there yet.
right now we can make rnodes with the sx1276 and sx1278 loraboards, support for  more chips is in the works,but is a big effort,hence supporting Mark Qvist is key.
all currently supported types of devices are in the supported hardware section.

**Comment by kc1awv** [comment]
> -(rnode): would this be the same as supporting any other sx1262 Lora board?

This is the important part, to me. From what I can see, the firmware as it stands right now (uses/needs/accesses?) the DIO0 pin, which is not present on the 1262. Maybe it's as simple as using a different library for the SX1262. I have a handful of Heltec LoRA boards (V3, with the ESP32-S3FN8 module) that I'd love to throw RNode on.

I may also purchase some of these 'sufficient' boards you've mentioned, myself.

**Comment by markqvist** [comment]
I've actually had a few prototype implementations up and running for SX1262 support, but they have so far all been inadequate in one or more ways.

There's some significant architectural differences between the SX1276 and the SX1262, which means that it's turning out a bit more complicated to get it all working than initially expected. Especially since it all needs to be backwards-compatible with everything, and that the system and firmware is already pretty complex ;)

We'll get there, that's for sure, but it is a pretty big effort. None of the existing libraries or drivers are good enough to base it on, so I am left with writing a completely new SX1262 raw interface driver for this, which really is a bit of an effort.

**Reply by cobraPA** [reply]
Hi @markqvist , how is this effort going, and is it something you can put up in development for others to work with you on?  I have interest in seeing some SX1262 boards work as well.

**Reply by markqvist** [reply]
Still more or less the same status as last. I only have so much time and resources, and other aspects of Reticulum development had to be prioritised higher.

The rationale for this is that, while SX1262 support will be nice to have, we already have a fully functional LoRa stack, based on SX127X chips, and other important parts of the overall ecosystem were missing essential functionality.
